pulp_service/pulp_service/tasking/new_worker.py

```diff
@@ -726,17 +709,34 @@ def handle_tasks(self):
                     # Release the task lock so other workers can attempt it
                     task_lock_key = f"task:{task.pk}"
                     self.redis_conn.delete(task_lock_key)
-            except Exception:
-                # Exception occurred - could be during is_compatible() check or before
-                # task execution begins.
+                    break
+
+                # Task is compatible, execute it
+                if task.immediate:
+                    self.supervise_immediate_task(task)
+                else:
+                    self.supervise_task(task)
+
+                # Release resource locks if subprocess didn't release them
+                # For deferred tasks running in subprocess, the subprocess doesn't have
+                # _locked_resources attribute, so locks aren't released there.
+                # For immediate tasks, _execute_task() releases locks and deletes this attribute.
+                if hasattr(task, '_locked_resources'):
+                    if task._locked_resources:
+                        self._release_resource_locks(task._locked_resources)
+                        _logger.debug(
+                            "Worker %s released resource locks for task %s after subprocess completion",
+                            self.name,
+                            task.pk
+                        )
+                    # Delete attribute to prevent finally block from trying to release again
+                    del task._locked_resources
```

Comment:
I don't think this needs to be here at all. I think the `finally` will do all the same stuff except the logging line. @dkliban wdyt?